tool/tsh/access_request.go

```diff
@@ -0,0 +1,308 @@
+				}
+			}
+			for _, reviewer := range req.GetSuggestedReviewers() {
+				if reviewer == cf.Username {
+					filtered = append(filtered, req)
+					continue Suggested
+				}
+			}
+		}
+		reqs = filtered
+	}
+	if cf.MyRequests {
+		filtered := reqs[:0]
+		for _, req := range reqs {
+			if req.GetUser() == cf.Username {
+				filtered = append(filtered, req)
+			}
+		}
+		reqs = filtered
+	}
+	switch cf.Format {
+	case teleport.Text:
+		if err := showRequestTable(reqs); err != nil {
+			return trace.Wrap(err)
+		}
+	case teleport.JSON:
+		ser, err := json.MarshalIndent(reqs, "", "  ")
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		fmt.Printf("%s\n", ser)
+	default:
+		return trace.BadParameter("unsupported format %q", cf.Format)
+	}
+	return nil
+}
+
+func onRequestShow(cf *CLIConf) error {
+	tc, err := makeClient(cf, false)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if cf.Username == "" {
+		cf.Username = tc.Username
+	}
+
+	var req types.AccessRequest
+	err = tc.WithRootClusterClient(cf.Context, func(clt auth.ClientI) error {
+		req, err = services.GetAccessRequest(cf.Context, clt, cf.RequestID)
+		return trace.Wrap(err)
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	reason := "none"
+	if r := req.GetRequestReason(); r != "" {
+		reason = fmt.Sprintf("%q", r)
+	}
+
+	reviewers := "none"
+	if r := req.GetSuggestedReviewers(); len(r) > 0 {
+		reviewers = strings.Join(r, ", ")
+	}
+
+	output := [][]string{
+		{
+			fmt.Sprintf("Request ID: %s", req.GetName()),
+			fmt.Sprintf("Username:   %s", req.GetUser()),
+			fmt.Sprintf("Roles:      %s", strings.Join(req.GetRoles(), ", ")),
+			fmt.Sprintf("Reason:     %s", reason),
+			fmt.Sprintf("Reviewers:  %s (suggested)", reviewers),
+			fmt.Sprintf("Status:     %s", req.GetState().String()),
+		},
+	}
+
+	joinBlocks := func(blocks [][]string, indent string) []string {
+		var maxlen int
+		for _, b := range blocks {
+			for _, l := range b {
+				if len(l) > maxlen {
+					maxlen = len(l)
+				}
+			}
+		}
+		sep := indent + strings.Repeat("-", maxlen-len(indent))
+
+		var out []string
+
+		for i, b := range blocks {
+			if i != 0 {
+				out = append(out, sep)
+			}
+			out = append(out, b...)
+		}
+		return out
+	}
+
+	var approvals, denials []types.AccessReview
+
+	for _, rev := range req.GetReviews() {
+		switch {
+		case rev.State.IsApproved():
+			approvals = append(approvals, rev)
+		case rev.State.IsDenied():
+			denials = append(denials, rev)
+		}
+	}
+
+	makeReviewBlock := func(title string, revs []types.AccessReview) []string {
+		const indent = "  "
+		blocks := [][]string{
+			{
+				fmt.Sprintf("%s:", title),
+			},
+		}
+		for _, rev := range revs {
+			revReason := "none"
+			if rev.Reason != "" {
+				revReason = fmt.Sprintf("%q", rev.Reason)
+			}
+			blocks = append(blocks, []string{
+				fmt.Sprintf("%sReviewer: %s", indent, rev.Author),
+				fmt.Sprintf("%sReason:   %s", indent, revReason),
+			})
+		}
+		return joinBlocks(blocks, indent)
+	}
```

Comment:
Adding so much ad hoc printing code looks like an overkill. Couldn't we take advantage of `asciitable` and print a [summary table](https://github.com/gravitational/teleport/blob/09bf1ca93124e8cb3525c5c42e89cd6e8859caec/tool/tctl/common/access_request_command.go#L351-L358) of the request info followed by a table with the columns like
```
Reviewer | Decision | Submitted (UTC) | Reason
```
showing the reviews?